src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -79,41 +55,25 @@ export default class Fms {
     /**
      * @constructor
      * @param aircraftInitProps {object}
-     * @param initialRunwayAssignment {RunwayModel}
      * @param typeDefinitionModel {AircraftTypeDefinitionModel}
      * @param navigationLibrary {NavigationLibrary}
      */
-    constructor(aircraftInitProps, initialRunwayAssignment, typeDefinitionModel, navigationLibrary) {
+    constructor(aircraftInitProps, typeDefinitionModel, navigationLibrary) {
         if (!_isObject(aircraftInitProps) || _isEmpty(aircraftInitProps)) {
             throw new TypeError('Invalid aircraftInitProps passed to Fms');
         }
 
         /**
-         * Instance of the `NavigationLibrary`
+         * Airport the aircraft arrives at
          *
-         * provides access to the aiport SIDs, STARs and Fixes via collection objects and fascade methods.
-         * used for route building
-         *
-         * @property _navigationLibrary
-         * @type {NavigationLibrary}
-         * @private
+         * @for Fms
+         * @property arrivalAirportModel
+         * @type {AirportModel}
          */
-        this._navigationLibrary = navigationLibrary;
-
-        /**
-        * routeSegments of legs that have been completed
-        *
-        * Used to generate a routeString for an entire, original, route
-        *
-        * @property _previousRouteSegments
-        * @type {array<string>}
-        * @default []
-        * @private
-        */
-        this._previousRouteSegments = [];
+        this.arrivalAirportModel = null;
```

Comment:
I think we've talked about this before, but I am _very_ wary of including an `AirportModel` instance as part of the `fms`.  I'm sure you have good reasons, but I feel like this is dangerous.